lib/src/code_generator/type.dart

```diff
@@ -48,6 +48,12 @@ abstract class Type {
   /// as getFfiDartType. For ObjC bindings this refers to the wrapper object.
   String getDartType(Writer w) => getFfiDartType(w);
 
+  /// Returns whether the FFI dart type and C type string are same.
+  bool get sameFfiDartAndCType;
```

Comment:
Maybe rename to `ffiDartAndCTypeEqual`. "Equal" seems a more logical word to use than "same".

Also, can we use `foo.getCType() == foo.getFfiDartType()` on the call-sites and avoid declaring this method? (I know we had this structure already before the PR, so feel free to punt to a separate refactoring.)